packages/runtime/src/components/page/Page.tsx

```diff
@@ -1,268 +1,35 @@
-  useEffect(() => {
-    const headSnippetsToCleanUp = (previousHeadSnippets.current ?? [])
-      .filter(previousSnippet => previousSnippet.cleanup != null)
-      .filter(previousSnippet => !headSnippets.some(snippet => previousSnippet.id === snippet.id))
-
-    headSnippetsToCleanUp.forEach(snippetToCleanUp => {
-      if (snippetToCleanUp.cleanup == null) return
-
-      const cleanUp = new Function(snippetToCleanUp.cleanup)
-
-      try {
-        cleanUp()
-      } catch {
-        // Ignore errors from user input.
-      }
-    })
-
-    previousHeadSnippets.current = headSnippets
-  }, [headSnippets])
-
   const documentId = baseLocalizedPage?.elementTreeId ?? page.id
 
   useRouterLocaleSync()
 
   return (
     <>
-      <Head>
-        <style>
-          {`
-            html {
-              font-family: sans-serif;
-            }
-            div#__next {
-              overflow: hidden;
-            }
-          `}
-        </style>
-
-        <link rel="icon" type={favicon.mimetype} href={favicon.publicUrl} />
-
-        {canonicalUrl && <link rel="canonical" href={canonicalUrl} />}
-
-        {isIndexingBlocked && <meta name="robots" content="noindex" />}
-
-        {title && (
-          <>
-            <title>{title}</title>
-            <meta property="og:title" content={title} />
-            <meta name="twitter:title" content={title} />
-            <meta itemProp="name" content={title} />
-          </>
-        )}
-
-        {description && (
-          <>
-            <meta name="description" content={description} />
-            <meta property="og:description" content={description} />
-            <meta name="twitter:description" content={description} />
-            <meta itemProp="description" content={description} />
-          </>
-        )}
-
-        {keywords && <meta name="keywords" content={keywords} />}
-
-        {socialImage && (
-          <>
-            <meta property="og:image" content={socialImage.publicUrl} />
-            <meta property="og:image:type" content={socialImage.mimetype} />
-            <meta name="twitter:image" content={socialImage.publicUrl} />
-            <meta name="twitter:card" content="summary_large_image" />
-            <meta itemProp="image" content={socialImage.publicUrl} />
-          </>
-        )}
-
-        {fontFamilyParamValue !== '' && (
-          <>
-            <link
-              rel="stylesheet"
-              href={`https://fonts.googleapis.com/css?family=${fontFamilyParamValue}&display=swap`}
-            />
-          </>
-        )}
-
-        {headSnippets.map(snippetToElement).map(children =>
-          Children.map(children, child => {
-            if (typeof child === 'string') return child
-
-            if (VALID_HEAD_ELEMENT_TYPES.includes(child.type as string)) return child
-
-            return null
-          }),
-        )}
-      </Head>
+      {/* Can also pass children to be added to page <head> here */}
```

Comment:
I'm not sure we should add this to the API. `useServerInsertedHTML` probably doesn't do anything useful if `children` changes and so having an API that doesn't work as expected might just make things more confusing. Given how differently `next/head` and `useServerInsertedHTML` behave we should probably not try to build some sort of useful abstraction for them as much as just hiding them behind `PageHead` and making sure the behavior works well for the exact use case `PageHead` tries to accomplish. There's already some mismatch there with cleaning up head scripts on client-side navigations that we need to potentially address.